src/services/etherspot.js

```diff
@@ -757,6 +758,24 @@ export class EtherspotService {
     }
   }
 
+  async getNftList(
+    chain: Chain,
+    address: string,
+  ): NftList | null {
+    const sdk = this.getSdkForChain(chain);
+
+    if (!sdk) return null;
+
+    try {
+      return sdk.getNftList({
+        account: address,
+      });
+    } catch (error) {
+      reportErrorLog('EtherspotService getNftList failed', { chain, error });
+      return null;
+    }
```

Comment:
Any reason we're using `try / catch` here instead of using a .catch here? `try / catch` should be used as a final resort...

Reply:
cant return null on .catch

Reply:
Are you sure about that? I'm quite sure that if you return `null` inside a catch block, the `promise`d function would return `null`... or any other value that is `return`ed in the `catch` block...

Reply:
catch block as of now will return null but if its inside .catch, its basically inside a callback function and .catch return type is void so thats y i think it wont return to the calling place

Reply:
I've just been doing a bit of research on this because Typescript inferring a `void` return type seems like a very opinionated action within a function (`.catch`) that has always been able to `return` anything that the developer wants to... i know that if you don't return anything inside a `.catch` block, Typescript adds a `void` union to the block itself but that's if you don't `return` anything. @Yash-Chaubal might you know anything about this scenario?

Reply:
i guess i was wrong abt that just now searched and yeah it is possible to return anything. So will change it to .catch